### PR TITLE
fix: rollback to tendermint v0.34.21

### PR DESCRIPTION
**Pull request by 0xmuralik**
Closes: #37 

## Description

Rollback to tendermint by taking https://github.com/evmos/ethermint as reference

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: testutil/network/util.go [hunk 1]
```diff
@@ -41,34 +45,39 @@ func startInProcess(cfg Config, val *Validator) error {
 	}
 
 	app := cfg.AppConstructor(*val)
-	genDoc, err := types.GenesisDocFromFile(tmCfg.GenesisFile())
+	// genDoc, err := types.GenesisDocFromFile(tmCfg.GenesisFile())
```
Comment (i-norden) [inline testutil/network/util.go:48]: Let's remove these commented out lines

**Review by i-norden — DISMISSED** [review]
LGTM on first pass but some of the tests are failing which weren't before.

**Comment by i-norden** [comment]
Hi @0xmuralik, what is the status on this? Can you please get the remaining tests passing on this so that we can merge, thanks!

**Review by i-norden — COMMENTED** [review]
Hey @0xmuralik, looks good, but it's not really feasible for me to review a change set this large line-by-line. So I think what would help the most is if you can get the remaining github actions passing or provide an explanation for why it is okay for them not to be passing (in which case we eventually need to adjust the actions or tests so that they do all pass). Thanks!

**Comment by 0xmuralik** [comment]
Hey @i-norden, Most of the failing tests from gosec and lint are false postives and very few need to be addressed. We can add comments to ignore these errors. I suggesting addressing these in a different PR.
And the protobuf breaking test is just to mention the changes in proto files; nothing to worry about.

**Comment by i-norden** [comment]
@0xmuralik are the failing `test-solidity`s an artifact of the CI? 1-3 is passing but 2-3, and 3-3 are not.

**Comment by 0xmuralik** [comment]
> @0xmuralik are the failing `test-solidity`s an artifact of the CI? 1-3 is passing but 2-3, and 3-3 are not.

`RangeError: Maximum call stack size exceeded` same error appears on laconicd main and ethermint main. These were failing before too. May be there is a limit on compiled contracts we an use in a test or something like that.

**Review by i-norden — APPROVED** [review]
LGTM

**Pull request merged**